PresnyaHedonist/Controllers/FeedVC.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  FeedVC.swift
+//  PresnyaHedonist
+//
+//  Created by Alexandr L. on 7/13/21.
+//
+
+import UIKit
+
+class FeedVC: UIViewController {
+    
+    var places = [Place]()
+    var filteredPlaces = [Place]()
+    
+    let context = (UIApplication.shared.delegate as! AppDelegate).persistentContainer.viewContext
+    let searchController = UISearchController(searchResultsController: nil)
+    
+    @IBOutlet weak var tableViewFeed: UITableView!
+    
+    
+    var searchBarIsEmpty: Bool {
+        guard let text = searchController.searchBar.text else { return false }
+        return text.isEmpty
+    }
+    
+    var isFiltering: Bool {
+        let searchScopeFiltering = searchController.searchBar.selectedScopeButtonIndex != 0
+        return searchController.isActive && (!searchBarIsEmpty || searchScopeFiltering)
+    }
+    
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableViewFeed.delegate = self
+        tableViewFeed.dataSource = self
+        tableViewFeed.tableSeparator()
+        
+        configureController()
+        configureSearchController()
+        
+        fetchData()
+    }
+    
+    
+    func configureController() {
+        navigationController?.navigationBar.prefersLargeTitles  = true
+        navigationController?.navigationBar.titleTextAttributes = [NSAttributedString.Key.font: Fonts.vcHeads!]
+        title = VCTitles.feedTitle
+    }
+    
+    
+    func fetchData() {
+        do {
+            places = try context.fetch(Place.fetchRequest())
+            
+        } catch {
+            presentAlert(title: AlertTitle.error,
+                         message: Errors.fetchError)
+        }
+    }
+    
+    
+    func configureSearchController() {
+        searchController.searchResultsUpdater                   = self
+        searchController.searchBar.delegate                     = self
+        searchController.obscuresBackgroundDuringPresentation   = false
+        searchController.searchBar.placeholder                  = "Поиск..."
+        searchController.searchBar.scopeButtonTitles            = ["Все", "Ресторан", "Места", "Завтраки", "Бар"]
+        
+        navigationItem.searchController                         = searchController
+        definesPresentationContext                              = true
+    }
+    
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if tableViewFeed.indexPathForSelectedRow == nil {
```

Comment:
This `if` is essentially a `guard` (in that it escapes the scope if not satisfied).
I prefer guards here too since they guarantee that the scope is exited.

Reply:
Corrected) thanks!